Formula/lib/libdpp.rb

```diff
@@ -0,0 +1,27 @@
+class Libdpp < Formula
+  desc "C++ Discord API Bot Library"
+  homepage "https://github.com/brainboxdotcc/DPP"
+  url "https://github.com/brainboxdotcc/DPP/releases/download/v10.0.28/DPP-10.0.28.tar.gz"
+  sha256 "aa0c16a1583f649f28ec7739c941e9f2bf9c891c0b87ef8278420618f8bacd46"
+  license "Apache-2.0"
+
+  depends_on "cmake" => :build
+  depends_on "libsodium"
+  depends_on "openssl@3"
+  depends_on "opus"
+  depends_on "pkg-config"
+
+  def install
+    system "cmake", "-B", "build", *std_cmake_args
+    cd "build" do
+      system "make"
+      system "make", "install"
+    end
+  end
+
+  test do
+    cd "build" do
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Our unit tests do exactly this however I've linked it incorrectly.

I'll fix it upstream (in D++) and we can await 10.0.29's release.

Reply:
Unless people regularly run the test suite as normal usage, that doesn't count

Reply:
> Unless people regularly run the test suite as normal usage, that doesn't count

No they don't, but I've already explained we can't do a normal regular user-case test as it would require a Discord token or a Discord Webhook. We can't just throw that into the formula as discord will keep resetting it as it's a security risk for me.

Reply:
> Unless people regularly run the test suite as normal usage, that doesn't count

No they don't, but I've already explained we can't do a normal regular user-case test as it would require a Discord token or a Discord Webhook. We can't just throw that into the formula as discord will keep resetting it as it's a security risk for me.

Reply:
Sounds like this piece of my text:
> If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.

Reply:
> Sounds like this piece of my text:
> 
> > If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.

Yeah, sounds good :)